src/main/kotlin/uk/gov/gdx/datashare/services/EventDataService.kt

```diff
@@ -119,20 +120,24 @@ class EventDataService(
     val enrichmentFields =
       acquirerSubscriptionEnrichmentFieldRepository.findAllByAcquirerSubscriptionId(subscription.acquirerSubscriptionId)
         .map { it.enrichmentField }
+
     return EventNotification(
       eventId = event.id,
       eventType = subscription.eventType,
       sourceId = event.dataId,
       dataIncluded = if (!callbackEvent) includeData else null,
-      enrichmentFields = if (!callbackEvent) enrichmentFields else null,
-      eventData = if (includeData) {
-        deathNotificationService.getEnrichedPayload(
-          event.dataId,
-          enrichmentFields,
-        )
-      } else {
-        null
-      },
+      enrichmentFields = enrichmentFields,
+      eventData = if (includeData) callbackAndEnrichData(subscription, event, enrichmentFields) else null,
     )
   }
+
+  private fun callbackAndEnrichData(
+    subscription: AcquirerSubscription,
+    event: EventData,
+    enrichmentFields: List<String>,
+  ): Any? = when (subscription.eventType) {
```

Comment:
Perhaps not now, but we should find a way to type this better

Reply:
Agree that we should later on